src/biogeochem/test/DustEmis_test/test_DustEmisZender2003.pf

```diff
@@ -0,0 +1,483 @@
+                     this%water_inst%waterdiagnosticbulk_inst, this%frictionvel_inst)
+   call this%dust_emis%DustDryDep(bounds, this%atm2lnd_inst, this%frictionvel_inst)
+   do p = bounds%begp, bounds%endp
+      call this%dust_emis%GetPatchVars( p, flx_mss_vrt_dst_tot=flx_mss_vrt_dst_tot )
+      @assertEqual( flx_mss_vrt_dst_tot, 0.0_r8 )
+   end do
+
+  end subroutine dust_zero_when_fsno_one
+
+  !-----------------------------------------------------------------------
+
+ @Test
+ subroutine dust_zero_non_veg_lu(this)
+   ! Check dust emissions are zero for non-veg landunits
+   use landunit_varcon, only: istcrop, max_lunit
+   use LandunitType, only : lun
+   class(TestDustEmisZender2003), intent(inout) :: this
+   integer :: p, l
+   real(r8) :: flx_mss_vrt_dst_tot
+
+   call this%create_atm2lnd()
+   call this%create_fv( )
+   ! Set the lanunit type for
+   do l = istcrop+1, max_lunit
+      lun%itype(bounds%begl:bounds%endl) = l
+      call this%dust_emis%DustEmission(bounds, this%num_nolakep, this%filter_nolakep, this%atm2lnd_inst, &
+                        this%soilstate_inst, this%canopystate_inst, this%water_inst%waterstatebulk_inst, &
+                        this%water_inst%waterdiagnosticbulk_inst, this%frictionvel_inst)
+      call this%dust_emis%DustDryDep(bounds, this%atm2lnd_inst, this%frictionvel_inst)
+      do p = bounds%begp, bounds%endp
+         call this%dust_emis%GetPatchVars( p, flx_mss_vrt_dst_tot=flx_mss_vrt_dst_tot )
+         @assertEqual( flx_mss_vrt_dst_tot, 0.0_r8 )
+      end do
+   end do
+
+  end subroutine dust_zero_non_veg_lu
+
+  !-----------------------------------------------------------------------
+
+  @Test
+  subroutine aborts_on_bad_dust_mobility(this)
+    ! Check dust abort when dust mobility is bad
+    class(TestDustEmisZender2003), intent(inout) :: this
+    real(r8) :: flx_mss_vrt_dst_tot
+    character(100) :: expected_msg
+ 
+    call this%create_atm2lnd()
+    call this%create_fv( )
+    ! Dust should abort with an error on dust mobility when snow fraction greater than 1
+    this%water_inst%waterdiagnosticbulk_inst%frac_sno_col(:) = 1.0_r8 + 1.e-15_r8
+    call this%dust_emis%DustEmission(bounds, this%num_nolakep, this%filter_nolakep, this%atm2lnd_inst, &
+                         this%soilstate_inst, this%canopystate_inst, this%water_inst%waterstatebulk_inst, &
+                         this%water_inst%waterdiagnosticbulk_inst, this%frictionvel_inst)
+    expected_msg = "ABORTED: Bad value for dust mobilization fraction"
+    @assertExceptionRaised(expected_msg)
+
+  end subroutine aborts_on_bad_dust_mobility
+ 
+  !-----------------------------------------------------------------------
+
+ @Test
+ subroutine dust_zero_when_tlai_high(this)
+   ! Check dust emissions are zero when LAI is high enough
+   class(TestDustEmisZender2003), intent(inout) :: this
+   integer :: p
+   real(r8) :: flx_mss_vrt_dst_tot
+
```

Comment:
Add a simple change to change the patch type (by overwritting itype) to something different than baresoil. And add some comments that to do this properly it might be best to clean everything and redo the setup starting with that veg type to begin with for this test (that could matter for potential changes down the road).

Reply:
Got it.